test-suite/shared/src/test/scala/org/scalajs/testsuite/javalib/util/CollectionsTest.scala

```diff
@@ -14,20 +14,50 @@ import org.junit.Test
 
 import org.scalajs.testsuite.utils.AssertThrows._
 import org.scalajs.testsuite.utils.CollectionsTestBase
-import org.scalajs.testsuite.utils.Platform.executingInJVM
 
 import scala.collection.JavaConversions._
 
 class CollectionsTest extends CollectionsTestBase {
 
+  private def checkImmutablilityOfCollectionApi[E](coll: ju.Collection[E],
+      elem: E): Unit = {
+    expectThrows(classOf[UnsupportedOperationException], coll.add(elem))
+    expectThrows(classOf[UnsupportedOperationException], coll.addAll(List(elem)))
+    expectThrows(classOf[Exception], coll.remove(elem))
+    expectThrows(classOf[Exception], coll.removeAll(List(elem)))
+    expectThrows(classOf[Exception], coll.retainAll(List(elem)))
+    expectThrows(classOf[Throwable], coll.clear())
+  }
+
+  private def checkImmutablilityOfCollectionApi[E](set: ju.Set[E],
+      elem: E): Unit = {
```

Comment:
I'm always wary of overloads where a parameter has a supertype of another overload. Consider renaming this to `checkImmutabilityOfSetApi`. Same remark for the other two overloads below.